src/pages/Profile/hooks/useProfile.ts

```diff
@@ -0,0 +1,15 @@
+import useAuthUserStore from "@/stores/useAuthUserStore"
+import { useMemo } from "react"
+
+interface profileProps {
+  userId?: string
+}
+
+const useProfile = ({ userId }: profileProps) => {
+  const { user } = useAuthUserStore()
+  const isMyPage = useMemo(() => userId === user._id, [userId, user._id])
+
+  return { isMyPage }
```

Comment:
isMyPage가 useMemo로 boolean값을 받고 있고 현재는 isMyPage만 return 하기 때문에 이렇게 바꿔보셔도 괜찮을 것 같아요! 😊😊
```suggestion
  const isMyPage = useMemo(() => userId === user._id, [userId, user._id])

  return isMyPage
```
이렇게 하신다면 Profile.tsx 에서는 이렇게 바꾸시면 될 것 같습니다!
```
const isMyPage = useProfile({ userId: id })
```